spec/models/reservation_spec.rb

```diff
@@ -577,5 +577,33 @@
     end
   end
 
+  context 'check in check out validations' do
+    before do
+      @res = FactoryGirl.create(:valid_reservation,
+                                start_date: Time.zone.today,
+                                due_date: Time.zone.today + 1.day)
+    end
+    it 'should fail to edit with check in time before start date' do
+      @res.update_attributes(checked_in: Time.zone.now - 2.days)
```

Comment:
You can use `assign_attributes` instead of `update_attributes` to avoid the database hit and speed things up. This is true for all of the tests below.